### PR TITLE
Animals move when you try to build

### PR DESCRIPTION
**Pull request by Whiteknight**
If you hover a building ghost over an animal, that animal will try to move out of your way.
Only sheep and zebras move. Wolves don't care about you or your city.
Animals try to move to a randomly selected adjacent offset.

IN THE FUTURE: I don't do anything with the formation. It really might be a good idea to re-form the formation so animals don't get scattered away from each other too much. Also in the future it would be nice if zebras ran farther away, they don't seem like they're the type for minimal movements.

### CODE REVIEW

**Review by crudelios — CHANGES_REQUESTED** [review]
The proof of concept looksa good, so far there's only a small code formatting change.

If you want to easily make a single animal's movement somewhat random, use `map_random_get(grid_offset)` and limit the result to the possible directions.  That way each tile will move the animal in a different way.

Also, this seems to only work for 1x1 buildings. If the animal is in the middle of a 5x5, it will only move one tile and will likely keep blocking. Ideally the entire building size is considered.

File: src/widget/city_building_ghost.c [hunk 1]
```diff
@@ -453,8 +454,20 @@ static void draw_default(const map_tile *tile, int x_view, int y_view, building_
...
-        if (fully_blocked || forbidden_terrain || (map_has_figure_at(tile_offset) && type != BUILDING_PLAZA)) {
+
+        if (fully_blocked || forbidden_terrain)
+        {
+            blocked_tiles[i] = 1;
+        } else if (map_has_figure_at(tile_offset) && type != BUILDING_PLAZA) {
             blocked_tiles[i] = 1;
+            int figure_id = map_figure_at(tile_offset);
+            figure *f = figure_get(figure_id);
+            if (f->type == FIGURE_SHEEP || f->type == FIGURE_ZEBRA) {
```
Comment (Keriew) [inline src/widget/city_building_ghost.c:465]: I think it'd be better for the logical division to make animal fleeing a function in `figuretype/animal.c` and call it here.

Comment (crudelios) [reply src/widget/city_building_ghost.c:465]: Agreed

File: src/figuretype/animal.c [hunk 2]
```diff
@@ -254,6 +256,35 @@ void figure_wolf_action(figure *f)
...
+{
+    return map_terrain_is(grid_offset, TERRAIN_TREE | TERRAIN_ROCK | TERRAIN_WATER |
+        TERRAIN_BUILDING | TERRAIN_SHRUB );
+}
+
+void figure_animal_try_nudge_at(const map_tile *building_center_tile, int animal_tile_offset, int building_size)
+{
+    int figure_id = map_figure_at(animal_tile_offset);
+    figure *f = figure_get(figure_id);
+    if (f->type == FIGURE_SHEEP || f->type == FIGURE_ZEBRA) {
```
Comment (crudelios) [inline src/figuretype/animal.c:269]: I think the function shouldn't execute if the animal is already moving, right?

**Review by crudelios — CHANGES_REQUESTED** [review]
Overall I think it's looking better now, here's some changes though.

**Comment by crudelios** [comment]
@Whiteknight  since you mention it doesn't work for the hippodrome, does it work for warehouses?

**Comment by Whiteknight** [comment]
It does work for Warehouses.

**Comment by crudelios** [comment]
I think this doesn't work for ponds, bathhouses and docks either because you only added the relevant code to `draw_default`, which doesn't catch all buildings.

**Comment by Whiteknight** [comment]
My previous commit fixes this PR for other building types.

File: src/figuretype/animal.h [hunk 3]
```diff
@@ -2,6 +2,7 @@
 #define FIGURETYPE_ANIMAL_H
 
 #include "figure/figure.h"
+#include "map/point.h"
```
Comment (crudelios) [inline src/figuretype/animal.h:5]: I don't think you need this 😉

**Review by crudelios — CHANGES_REQUESTED** [review]
Just a very little change and I'll merge this!

File: src/figuretype/animal.h (hunk 3 above)
Comment (Whiteknight) [reply src/figuretype/animal.h]: Fixed and rebased.

**Comment by crudelios** [comment]
Thanks for this!

**Pull request merged**